packages/dd-trace/src/appsec/rasp/sql_injection.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+const { pgQueryStart, pgPoolQueryStart, pgPoolQueryFinish } = require('../channels')
+const { storage } = require('../../../../datadog-core')
+const addresses = require('../addresses')
+const waf = require('../waf')
+const { RULE_TYPES, handleResult } = require('./utils')
+
+let config
+
+function enable (_config) {
+  config = _config
+
+  pgQueryStart.subscribe(analyzePgSqlInjection)
+  pgPoolQueryStart.subscribe(analyzePgSqlInjectionInPool)
+  pgPoolQueryFinish.subscribe(pgPoolFinish)
+}
+
+function disable () {
+  if (pgQueryStart.hasSubscribers) pgQueryStart.unsubscribe(analyzePgSqlInjection)
+  if (pgPoolQueryStart.hasSubscribers) pgPoolQueryStart.subscribe(analyzePgSqlInjectionInPool)
+  if (pgPoolQueryFinish.hasSubscribers) pgPoolQueryFinish.subscribe(pgPoolFinish)
+}
+
+function analyzePgSqlInjection (ctx) {
+  const query = ctx.query?.text
+  if (!query) return
+
+  const store = storage.getStore()
+  if (!store) return
+
+  const { raspSqlAnalyzed, req } = store
+
+  if (!req || raspSqlAnalyzed) return
+
+  const persistent = {
+    [addresses.DB_STATEMENT]: query,
+    [addresses.DB_SYSTEM]: 'postgresql' // TODO: Extract to constant
+  }
+
+  const result = waf.run({ persistent }, req, RULE_TYPES.SQL_INJECTION)
+
+  const res = store?.res
+  handleResult(result, req, res, ctx.abortController, config)
+}
+
+function analyzePgSqlInjectionInPool (ctx) {
```

Comment:
pg.Pool internally calls to pg.Query, we could detect threats only listening to pg.Query, but the stacktrace that we have in pg.Client when we are using pg.Pool is bad, it is executed in other Stack and the information that we have is not important. The stacktrace when pool.Query is executed is more important that we have when pg.Query is executed under pool